### ISSUE TITLE
Sandbox: killed processes will not trigger the timeout handler

### ISSUE THREAD

**Issue by mikemaccana** [architect/sandbox#30]
Found as part of architect/architect#354.

Simple standalone reproduction case, based on `spawn.js`

Expect a PR shortly.

```
let spawn = require('child_process').spawn
var verySlowNodeCode = "setTimeout(function(){console.log(`Finished`)},20000)"
var log = console.log.bind(console)

var child = spawn('node', ['-e', verySlowNodeCode], { 
    shell: true,
    cwd: 'C:\\Users\\mikem\\OneDrive\\Documents\\appless\\documentdb-arc-test\\src\\http\\get-index',
    env: {}
})


let stdout = ''
let stderr = ''
let timeout = 5000;
// bake a timeout
let to = setTimeout(function () {
    log(`Timed out after ${timeout} ms!`)
    timedout = true
    child.kill()
}, timeout)

child.stdout.on('data', data => {
    log(`Got output data ${data}`)
    // always capture data piped to stdout
    // python buffers so you might get everything despite our best efforts
    stdout += data
})

child.stderr.on('data', data => {
    log(`Got error data ${data}`)
    stderr += data
})

child.on('close', function done (code) {
    log(`Closed!`)
    // Output any console logging from the child process
    
})

child.on('exit', function done (code) {
    log(`Exited!`)
    // Output any console logging from the child process
})
```

## What does happen

The process with `verySlowNodeCode` takes 20 seconds. It is killed after 5, however the checks for timeouts never run because `child.kill()` does not trigger `child.on('close'`

## What should happen

The process with `verySlowNodeCode` takes 20 seconds. It is killed after 5, and a timeout message should be returned to the user.

**Comment by ryanblock** [comment]
Good news, I think my fix for #33 will also fix this issue. Stand by!

**Comment by ryanblock** [comment]
Ok, fix for this is live in sandbox@RC! `npm i @architect/sandbox@RC` – please give it a whirl.

I'm happy with the changes, but I'm going to hold off merging the PR (https://github.com/architect/sandbox/pull/104) until we've verified the fix and put it through its paces.

**Comment by ryanblock** [comment]
Fix now live in Architect `6.0.15` and Sandbox `1.4.7`.
